gfs/chunkserver/services/chunkserver_servicer.go

```diff
@@ -50,6 +57,28 @@ func NewChunkServer(addr string) ChunkServer {
 		pendingTxsLock:    sync.Mutex{}}
 }
 
+// TODO: issue with wrapping around `NewChunkServer()` because Mutex is copy by value, not reference
+func NewBenchmarkingChunkServer(addr string, config BenchmarkConfig) ChunkServer {
+	fmt.Println("starting up chunkserver " + addr + ".")
+	chunkserverRootDir := chunkServerTempDirectoryPath + addr
+	if err := os.MkdirAll(chunkserverRootDir, os.ModePerm); err != nil {
+		// TO:DO This shouldn't really fatally crash the program, it's just one program
+		log.Fatal(err)
+	}
+
+	return ChunkServer{
+		ChunkHandleToFile: make(map[uint64]string),
+		Rootpath:          chunkserverRootDir,
+		Address:           addr,
+		WriteCache:        make(map[string]*protos.WriteDataBundle),
+		leases:            make(map[uint64]int64),
+		pendingChunkTxs:   make(map[uint64][]string),
+		completedTxs:      make(map[string]bool),
+		inProgressTxs:     make(map[string]bool),
+		pendingTxsLock:    sync.Mutex{},
+		BenchmarkConfig:   config}
```

Comment:
Unlike the standard `NewChunkServer()` function, this one adds the benchmarking config so we can toggle between Sequential and Async chunkServer forwarding.